### PR TITLE
Fix a race condition and a nullptr dereference

### PR DESCRIPTION
**Pull request by t0suj4**
Related to two bugs of #2913

### CODE REVIEW

File: src/environment.cpp [hunk 1]
```diff
@@ -1826,21 +1826,23 @@ void ServerEnvironment::deactivateFarObjects(bool force_delete)
 
 				MapBlock *block = m_map->emergeBlock(obj->m_static_block, false);
 
-				std::map<u16, StaticObject>::iterator n =
+				if (block) {
+					std::map<u16, StaticObject>::iterator n =
 						block->m_static_objects.m_active.find(id);
-				if(n != block->m_static_objects.m_active.end()){
-					StaticObject static_old = n->second;
+					if(n != block->m_static_objects.m_active.end()){
```
Comment (est31) [inline src/environment.cpp]: space

File: src/environment.cpp [hunk 2]
```diff
@@ -1826,21 +1826,23 @@ void ServerEnvironment::deactivateFarObjects(bool force_delete)
...
 
-					float save_movem = obj->getMinimumSavedMovement();
+						float save_movem = obj->getMinimumSavedMovement();
 
-					if(static_old.data == staticdata_new &&
-							(static_old.pos - objectpos).getLength() < save_movem)
-						data_changed = false;
-				} else {
-					errorstream<<"ServerEnvironment::deactivateFarObjects(): "
+						if(static_old.data == staticdata_new &&
```
Comment (est31) [inline src/environment.cpp]: space

**Comment by est31** [comment]
Commit messages should be in present tense.

**Comment by t0suj4** [comment]
no spaces on my side

**Comment by est31** [comment]
:+1: if the things i pointed out are fixed.

File: src/environment.cpp (hunk 1 above)
Comment (kilbith) [inline src/environment.cpp]: > no spaces on my side

est31 means `if[space](...)[space]{`

**Comment by t0suj4** [comment]
oh, those spaces

**Comment by est31** [comment]
67997af67fbf7c95e78574f810aa322275b3eb66
79e2647556773d46cc0299e73ca23c41a153ff2a

**Pull request closed without merging**